### DISCUSSION TITLE
Topic chosen vs Probability mismatch?

### DISCUSSION THREAD

**Discussion by noahberhe** [MaartenGr/BERTopic#1371] [Q&A] [answered]
Hello,

So I'm comparing the topic_id chosen vs. the max(prob) in the probs array.
Generally topic has the max(prob) in the probs array, however there are a handful of cases where it doesn't, and it appears Bertopic has chosen to assign a topic that was e.g. the 2nd highest probability in the probs array and not the 1st.

Why is this please?

Thanks

**Comment by MaartenGr** [comment, marked as answer]
The probabilities are directly retrieved from the cluster model which in this case is HDBSCAN. That model does not use its probabilities to do the assignment of clusters. They are calculated **after** cluster assignment and are an approximation of the probabilities. The documentation of HDBSCAN [here](https://hdbscan.readthedocs.io/en/latest/soft_clustering.html) and [here](https://hdbscan.readthedocs.io/en/latest/soft_clustering_explanation.html) explains that in more detail.
